traitsui/testing/tester/wx/implementation/enum_editor.py

```diff
@@ -0,0 +1,195 @@
+            control=wrapper.target.source.control,
+            index=wrapper.target.location.index,
+            delay=wrapper.delay))),
+    ]
+
+
+class _IndexedRadioEditor(_BaseSourceWithLocation):
+    """ Wrapper class for EnumRadioEditor and Index.
+    """
+    source_class = RadioEditor
+    locator_class = locator.Index
+    handlers = [
+        (command.MouseClick,
+            (lambda wrapper, _: helpers.mouse_click_radiobutton_child_in_panel(
+                control=wrapper.target.source.control,
+                index=convert_index(
+                    source=wrapper.target.source,
+                    index=wrapper.target.location.index
+                ),
+                delay=wrapper.delay))),
+    ]
+
+
+def convert_index(source, index):
+    """ Helper function to convert an index for a GridSizer so that the
+    index counts over the grid in the correct direction.  
+    The grid is always populated in row major order, however, the elements
+    are assigned to each entry in the grid so that when displayed they appear
+    in column major order.
+    Sizers are indexed in the order they are populated, so to access
+    the correct element we may need to convert a column-major based index
+    into a row-major one.
+
+    Parameters
+    ----------
+    control : RadioEditor
+        The RadioEditor of interest. Its control is the wx.Panel containing
+        child objects organized with a wx.GridSizer
+    index : int
+        the index of interest
+    """
+    sizer = source.control.GetSizer()
+    if isinstance(sizer, wx.BoxSizer):
+        return index
+    n = len(source.names)
+    num_cols = sizer.GetCols()
+    num_rows = sizer.GetEffectiveRowsCount()
+    # the last entries of the last row can be empty.
+    num_empty_entries_last_row = num_cols * num_rows - n
+
+    # if the index of interest extends into the part of the grid where
+    # the columns have a missing entry in the last row 
+    if index > num_rows * (num_cols - num_empty_entries_last_row):
+        # break the grid up into 2 grids.  One of size 
+        # num_rows x (num_cols - num_empty_entries_last_row).  The other
+        # of size (num_rows-1) x num_empty_entries_last_row
+        num_entries_grid1 = index - num_rows * (num_cols - num_empty_entries_last_row)
+        # find i, j coordinates of the index in grid2 if we counted in
+        # column major order
+        new_index = index - num_entries_grid1
+        i = new_index % (num_rows - 1)
+        j = new_index // (num_empty_entries_last_row)
+        # convert that back to an index found from row major order
+        return num_entries_grid1 + (i * num_empty_entries_last_row + j)
+    else:
+        # find i,j coordinates of index if we counted in column major order
+        i = index % num_rows
+        j = index // num_rows
+    # convert that back to an index found from row major order
+    return i * num_cols + j
```

Comment:
this logic is repeated on Qt as well.  I am thinking I might refactor by pulling this out into a helper function that could be kept in a `utils` module or something like that.  The code is a bit complicated, I tried to make comments to make it more clear.  The code for populating the QGridLayout and the wx.GridSizer is complicated itself.  (I find it strange that the for loops for populating them are based on going in row order, but the default/ standard behavior is to use column order.  This forces the use of the `incr` array and for the population to occur in a row major order, but with the column major ordered elements) see https://github.com/enthought/traitsui/blob/81902cc48f0af498d4f0f1a980dc8074da4d8265/traitsui/qt4/enum_editor.py#L334 and https://github.com/enthought/traitsui/blob/81902cc48f0af498d4f0f1a980dc8074da4d8265/traitsui/wx/enum_editor.py#L357

Reply:
One thing to note is that this code will likely be applicable to checklist editor as well i realized so the module `traitsui.testing.tester.editors.enum_editor` might be too specific

Reply:
Hmmm, `layout` instead of `enum_editor`?

Reply:
(Eventually I think it should really be given a private module name, so if we regret the name later there is less to worry about. But let's wrap up all the pending PRs first before doing the overall naming changes. See first item of #1173 )